File: 프로그래머스/lv1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
    let answer ='';
    for(let i=0; i<s.length; i++){
        let temp = s[i].charCodeAt();
        if(s[i] === ' '){
            answer+=' ';
            continue;
        }
        if(96 < temp ){
            if(temp+n > 122){
                answer+=String.fromCharCode((((temp+n)%122)+96));
            }else{
                answer+=String.fromCharCode(temp+n);
            }
        }else{
            if(temp+n > 90){
               answer+=String.fromCharCode((((temp+n)%90)+64));
               }else{
                answer+=String.fromCharCode(temp+n);
            }
        }
     }
    return answer
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
import java.util.*;

class Solution {
    public String solution(String s, int n) {
        
        String answer = "";
        int leng = s.length();
        
        for(int i=0; i<leng; i++){
            char ch = s.charAt(i);
            
            if(Character.isWhitespace(ch)){
                answer += " ";
                continue;
            }
            
            char base = Character.isUpperCase(ch) ? 'A' : 'a';
            
            int cur = (ch - base + n)%26 + base;
            
            answer += (char)cur;
        }
        
        return answer;
    }
}